src/report.rs

```diff
@@ -190,6 +190,78 @@ impl Report for TapReport {
     fn done(&mut self) {}
 }
 
+/// This is a particularly simpl reporter, mostly useful with --nocapture,
+/// that will collect stats and results and will display a summary
+/// after all tests are done.
+pub struct CompactReport {
+    writer: ColorWriter,
+    results: BTreeMap<String, (Duration, Status)>,
+    stats: TestStats,
+}
+
+impl CompactReport {
+    pub fn new(writer: ColorWriter) -> Self {
+        Self {
+            writer,
+            results: BTreeMap::new(),
+            stats: Default::default(),
+        }
+    }
+}
+
+impl Report for CompactReport {
+    fn init(&mut self, _plan: &[Task]) {}
+
+    fn start(&mut self, _task_name: String) {}
+
+    fn report(&mut self, result: &CompletedTask) {
+        self.stats.update(result);
+        self.results.insert(
```

Comment:
Is it really that important to have test labels sorted? I assume most users will prefer interactive output to waiting until the very last test completes just in order to see the test labels sorted...

Reply:
But when using `--format compact --nocapture` you'll see the interactive output. I really want a list of all executed tests at the end. Precisely so its easy to see what happened after a long test run.

If the users prefer, they can stick with `--format libtest` for a more familiar report format.

Reply:
But then the test logs will form one huge blob and it will be hard to find where one test begins and the other test ends. Also note that libtest format explicitly enumerates all the failed test in the summary, which is what people should probably care about the most.